dev/com.ibm.ws.jpa_spec10_injection_fat/test-applications/injection_dpu/src/com/ibm/ws/jpa/fvt/injectiondpu/web/applevel/jta/AppLevelJTADPUFieldInjectionServlet.java

```diff
@@ -0,0 +1,93 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.jpa.fvt.injectiondpu.web.applevel.jta;
+
+import java.util.HashMap;
+
+import javax.annotation.PostConstruct;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+
+import org.junit.Test;
+
+import com.ibm.ws.jpa.fvt.injectiondpu.testlogic.InjectionDPUTestLogic;
+import com.ibm.ws.testtooling.testinfo.JPAPersistenceContext;
+import com.ibm.ws.testtooling.testinfo.JPAPersistenceContext.PersistenceContextType;
+import com.ibm.ws.testtooling.testinfo.JPAPersistenceContext.PersistenceInjectionType;
+import com.ibm.ws.testtooling.testinfo.TestExecutionContext;
+import com.ibm.ws.testtooling.vehicle.web.JPATestServlet;
+
+@SuppressWarnings("serial")
+//@WebServlet(urlPatterns = "/AppLevelJTADPUFieldInjectionServlet")
+public class AppLevelJTADPUFieldInjectionServlet extends JPATestServlet {
+    @SuppressWarnings("unused")
+    @PersistenceContext
+    private EntityManager cmtsEM;
+
+    @SuppressWarnings("unused")
+    @PersistenceUnit
+    private EntityManagerFactory amjtaEMF;
+
+    private final String testLogicClassName = InjectionDPUTestLogic.class.getName();
+
+    private final HashMap<String, JPAPersistenceContext> jpaPctxMap = new HashMap<String, JPAPersistenceContext>();
+
+    @PostConstruct
+    private void initFAT() {
+        jpaPctxMap.put("test-jpa-resource-amjta",
+                       new JPAPersistenceContext("test-jpa-resource-amjta", PersistenceContextType.APPLICATION_MANAGED_JTA, PersistenceInjectionType.FIELD, "amjtaEMF"));
+
+        jpaPctxMap.put("test-jpa-resource-cmts",
+                       new JPAPersistenceContext("test-jpa-resource-cmts", PersistenceContextType.CONTAINER_MANAGED_TS, PersistenceInjectionType.FIELD, "cmtsEM"));
+    }
+
+    @Test
+    public void jpa10_Injection_DPU_AppLevel_Field_Web_AMJTA() throws Exception {
+        final String testName = "jpa10_Injection_DPU_AppLevel_Field_Web_AMJTA";
+        final String testMethod = "testDefaultPersistenceUnitInjection";
+
+        final TestExecutionContext testExecCtx = new TestExecutionContext(testName, testLogicClassName, testMethod);
+
+        final HashMap<String, JPAPersistenceContext> jpaPCInfoMap = testExecCtx.getJpaPCInfoMap();
+        jpaPCInfoMap.put("test-jpa-resource", jpaPctxMap.get("test-jpa-resource-amjta"));
+
+        HashMap<String, java.io.Serializable> properties = testExecCtx.getProperties();
+        properties.put("dbProductName", getDbProductName());
+        properties.put("dbProductVersion", getDbProductVersion());
+        properties.put("jdbcDriverVersion", getJdbcDriverVersion());
```

Comment:
Ive been wondering about the necessity of get `JdbcDriverVersion`. Its just the same value as the `dbproductversion`, right?

DatabaseManagementServlet.getInfo:
```
            properties.put("dbproduct_version", dbMeta.getDatabaseProductVersion());
            properties.put("jdbcdriver_version", dbMeta.getDatabaseProductVersion());
```

Reply:
We don't presently have anything that uses it.  However, there may come a time where we may need it in the future, and having it available causes no harm.